orchagent/routeorch.cpp

```diff
@@ -2136,7 +2209,9 @@ bool RouteOrch::addRoutePost(const RouteBulkContext& ctx, const NextHopGroupKey
         {
             // Previous added an temporary route
             auto& tmp_next_hop = ctx.tmp_next_hop;
-            addRoutePost(ctx, tmp_next_hop);
+            if (tmp_next_hop.getSize() > 0) {
```

Comment:
is this change related to SRv6? any issues with regular flow?

Reply:
Looks like it is to avoid the situation that `addRoutePost()` is invoked twice while `addRoute()` is called only once. For example, `addTempRoute()` called by `addRoute()` returns false before invoking `addRoute()` again, so the size of `ctx.tmp_next_hop` is zero, which is constructed by default. Without the condition `if (tmp_next_hop.getSize() > 0)`, another `addRoutePost()` function call will be made, even though the `addRoute()` is called just once.